packages/react/src/components/MultiSelect/FilterableMultiSelect.js

```diff
@@ -184,17 +190,20 @@ export default class FilterableMultiSelect extends React.Component {
     }
   };
 
-  handleOnToggleMenu = () => {
+  handleOnMenuChange = (isOpen) => {
     this.setState((state) => ({
-      isOpen: !state.isOpen,
+      isOpen: isOpen ?? !state.isOpen,
     }));
+    if (this.props.onMenuChange) {
+      this.props.onMenuChange(isOpen);
+    }
   };
 
   handleOnOuterClick = () => {
     this.setState({
-      isOpen: false,
       inputValue: '',
     });
+    if (this.state.isOpen) this.handleOnMenuChange(false);
```

Comment:
```suggestion
```

I believe you can eliminate this line to avoid the event firing twice on close, since you are already firing it on state change

Reply:
Yep, that worked perfectly, thanks!